### DISCUSSION TITLE
semicolons as function argument separators?

### DISCUSSION THREAD

**Discussion by unmovingcastle** [typst/typst#4917] [Q&A] [answered]
Hi,
Currently the physica `braket` function works like so:
`braket(a,b)` produces $\langle  a |  b \rangle $
However, I need to type a comma in both fields:
$\langle a,b | c=d, e=f \rangle$ 

I was therefore wondering if I could somehow write a custom function that uses a semicolon to separate these two? Something like `braket(a,b ; c=d, e=f)`

I know that the matrix function `mat` implements this somehow, but I can't find the source code for that.

My current workaround is to define
`#let comma=","`
so that
`braket(a #comma b, whatever)`
produces the desired $\langle a,b | \rm{whatever} \rangle$, but this is very ugly and inconvenient.

Could someone point me in the right direction? Many thanks in advance!

**Comment by laurmaedje** [comment, marked as answer]
mat uses both: comm and semicolon. This syntax is automatically available for all math function calls and produces array arguments.

Apart from that you can escape the comma with a backslash: `\,`

**Reply by unmovingcastle** [reply]
Ohhh I totally didn't even think about escaping hehehe.

So would the function look something like
`#let my_fun(array1, array2)=...`?

**Reply by laurmaedje** [reply]
Yeah exactly
